plugins/products.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Store} from 'redux';
+
+import store from 'stores/redux_store';
+
+import PluginRegistry from './registry';
+
+export abstract class ProductPlugin {
+    abstract initialize(registry: PluginRegistry, store: Store): void;
+    abstract uninitialize(): void;
+}
+
+export function initializeProducts() {
+    /* eslint-disable no-console, import/no-unresolved */
+    return async (/*dispatch, getState*/) => {
+        /**
+         * products contains a map of product IDs to a function that will load all of their parts. Calling that
+         * function will return an object where each field is a Promise that will resolve to that module.
+         *
+         * Note that these import paths must be statically defined or else they won't be found at runtime.
+         *
+         * Since this file doesn't use TypeScript yet, the type definition for it is:
+         *
+         * const products: Array<{
+         *     id: string;
+         *     load: () => {
+         *         index: Promise<{default: ProductPlugin}>;
+         *         manifest: Promise<{default: PluginManifest}>;
+         *     };
+         * }>;
+         */
+        const products = [
+            {
+                id: 'focalboard',
+                load: () => ({
+                    index: import('focalboard'),
```

Comment:
These dynamic imports return a promise that Webpack sets up to import the remote module from the location defined in webpack.config.js. Each `import` call has to take a literal string as a path for Webpack to be able to properly set everything up.

Each of them corresponds to one of the modules exposed by the product [using the exposes option of the ModuleFederationPlugin](https://github.com/mattermost/focalboard/pull/3748/files#diff-1c5589d2b7076ec8b90835ff95b065e6f57d7330758e104e1f853e079368c7c2R150)